File: 2016 - La Sapienza/TDP/Esercitazioni_TDP/Eserc3/es3.2.c
/*	CONSEGNA
Scrivere un programma che dati in input due numeri interi positivi
N ed M (M sottomultiplo di N), disegni una scacchiera
N*MxN*M con blocchi di asterischi M*M. 
 */

#include <stdio.h>

int main () {
	int m = -1; // height
	int n = -1; // width
	int c = 0;	//
	int count1=0;
    int count2=0;
    int count3=0;
    int count=0;
	int booleana = 0; //0 is false; 1 is true
	while ( !(m>0 && n>0 && booleana && m!=1)) {
		printf("Enter 2 positive number: ");
		scanf("%d%d", &n, &m);
		if (n%m==0) {
			booleana = 1;
		}
	}
	c = (m*n)/(2*m);
	count3=0;
	while (count3!=c) {			//Ripete la prima e la seconda serie per c volte
		count=0;
		while (count!=m) {		//Completa la prima serie che inizia con *
			count1=0;
			while (count1!=c) {	//Formato riga che inizia con *
				count2=0;
				while (count2!=m) {
					printf("*");
					count2++;
				}
				count2=0;
				while (count2!=m) {
					printf("-");
					count2++;
				}
				count1++;
			}
			count++;
			printf("\n");
		}
		count=0;
		while (count!=m) {		//Completa la prima serie che inizia con -
			count1=0;
			while (count1!=c) {	//Formato riga che inizia con -
				count2=0;
				while (count2!=m) {
					printf("-");
					count2++;
				}
				count2=0;
				while (count2!=m) {
					printf("*");
					count2++;
				}
				count1++;
			}
			count++;
			printf("\n");
		}
		count3++;
	}
	return 0;
}

File: 2016 - La Sapienza/TDP/Esercitazioni_NEW/Eserc3/es3.2.py
#	CONSEGNA
#Scrivere un programma che dati in input due numeri interi positivi
#N ed M (M sottomultiplo di N), disegni una scacchiera
#N*MxN*M con blocchi di asterischi M*M.

print("Inserisci numeri interi positivi\nIl 1° deve essere un multiplo del 2°")
n = int(input("Inserisci primo numero:\t"))
m = int(input("Inserisci un sottomultiplo del primo: "))
right = n%m==0

while not(m>0 and n>0 and right and m!=1):
    n = int(input("Inserisci primo numero:\t"))
    m = int(input("Inserisci un sottomultiplo del primo: "))

S = "*" #Star
L = "-" #Line
c = int((m*n)/(2*m))

for i in range(0,c):
    for i1 in range(0,m):
        print((S*m+L*m)*c)
    for i2 in range(0,m):
        print((L*m+S*m)*c)

#m=2
#n=4
#**--**--
#**--**--
#--**--**
#--**--**
#**--**--
#**--**--
#--**--**
#--**--**
